lms/urls.py

```diff
@@ -714,36 +733,11 @@
             ),
             include('lms.djangoapps.discussion.django_comment_client.urls')
         ),
-        url(
-            r'^notification_prefs/enable/',
-            notification_prefs_views.ajax_enable
-        ),
-        url(
-            r'^notification_prefs/disable/',
-            notification_prefs_views.ajax_disable
-        ),
-        url(
-            r'^notification_prefs/status/',
-            notification_prefs_views.ajax_status
-        ),
-        url(
-            r'^notification_prefs/unsubscribe/(?P<token>[a-zA-Z0-9-_=]+)/',
-            notification_prefs_views.set_subscription,
-            {
-                'subscribe': False,
-            },
-            name='unsubscribe_forum_update',
-        ),
-        url(
-            r'^notification_prefs/resubscribe/(?P<token>[a-zA-Z0-9-_=]+)/',
-            notification_prefs_views.set_subscription,
-            {
-                'subscribe': True,
-            },
-            name='resubscribe_forum_update',
-        ),
     ]
 
+if is_forum_notifications_enabled():
+    urlpatterns += notification_prefs_urls
```

Comment:
URLs are initialized at startup, meaning that changing a waffle flag would still require the process to restart before the URLs become active?

Is there a reason why this isn't implemented as a Django setting?

Reply:
No particular reason, i just wanted to use the one method for enabling/disabling the feature. 
Thanks for adding the context. 

If I add FEATURE settings (e.g. `settings.FEATURES('NOTIFICATION_PREF', is_forum_notifications_enabled())` that code is also initialized at startup and I am not sure making a DB call at that time would be the right thing to do.  

Another option is `settings.FEATURES('NOTIFICATION_PREF', True)` which would require adding the value in edx-internal.
What other ways we can achieve that?

Reply:
If it's important to keep it as a WaffleFlag (e.g. you need to enable it on per-course basis, there are backwards compatibility issues to consider, etc.), then you can always include the URLs but do the check when the view is invoked. If it's not important to keep it as a WaffleFlag and it's a system-wide enable/disable toggle, then please put it in settings (and thus edx-internal). We used to favor Waffle for some settings because of how quickly we could change the value, but doing it via settings is much faster than it used to be and has better auditing. Waffle should be used for selective rollout cases.

The one thing I want to avoid is using something that changes dynamically (e.g. waffle flag, configmodel) to toggle behavior that only occurs at startup time because that will lead to confusing situations where people have toggled the switch but don't see the behavior they expect.

Reply:
Right, Copy that. Thank you for the explaination. :)

I will move forward with the settings/feature/edx-internal settings.